src/main/java/backend/RepoId.java

```diff
@@ -0,0 +1,100 @@
+package backend;
+
+import org.eclipse.egit.github.core.RepositoryId;
+import util.Utility;
+
+/**
+ * Represents the ID of a repository. To be used to ensure safe comparison
+ */
+public final class RepoId {
+
+    private final String repoOwner;
+    private final String repoName;
+
+    /**
+     * Constructor
+     * @param repoIdString repoOwner and repoName separated by slash
+     * @throws IllegalArgumentException If repoIdString is invalid
+     */
+    public RepoId(String repoIdString) throws IllegalArgumentException {
+        boolean isWellFormed = RepoId.isWellFormedRepoIdString(repoIdString);
+        throwExceptionIfNotWellFormed(isWellFormed);
+
+        String[] repoIDComponents = repoIdString.toLowerCase().split("/");
+        this.repoOwner = Utility.removeAllWhitespace(repoIDComponents[0]);
+        this.repoName = Utility.removeAllWhitespace(repoIDComponents[1]);
+    }
+
+    /**
+     * Constructor
+     * @param repoOwner Owner of the repo
+     * @param repoName Name of the repo
+     * @throws IllegalArgumentException If repoOwner or repoName is invalid
+     */
+    public RepoId(String repoOwner, String repoName) throws IllegalArgumentException {
+        boolean isWellFormed = RepoId.isWellFormedRepoIdString(repoOwner, repoName);
+        throwExceptionIfNotWellFormed(isWellFormed);
+
+        this.repoOwner = Utility.removeAllWhitespace(repoOwner.toLowerCase());
+        this.repoName = Utility.removeAllWhitespace(repoName.toLowerCase());
+
+    }
```

Comment:
To reduce code (and duplication), you can remove this constructor altogether. i.e. have only one constructor that takes the full repo name as a single string.